### DISCUSSION TITLE
Unable to get desired behaviour by using reset button/reset command

### DISCUSSION THREAD

**Discussion by NsinghP** [arendst/Tasmota#11645] [Q&A] [answered]
Hello, 

Upon reset, I simply want the device to take the user to Tasmota access point where wifi settings can be configured afresh. I have compiled a custom bin file with desired configuration settings. This bin file is working as far as custom settings are concerned, but when:

(1)	I  Issue the reset command it hangs on to the old wifi settings and does not revert to tasmota access point mode to configure wifi settings.

(2)	(a)  When I press the reset button (Connected between GND and RST pin) the device simply reboots.
	(b)	 If I press reset button  six times, the device fallbacks to sonoff basic template and l have to manually select my custom template again. Device still retains wifi settings.

The Question:

How do I get my device to prompt for fresh wifi settings when I issue the reset command or press the reset button?

Regards,

Navi

**Comment by barbudor** [comment, marked as answer]
Could you please share in details what you've done ?

**Reply by barbudor** [reply]
You are misunderstanding the RESET pin of the ESP.
RESET button/pin is intended to reset the processor, which in terms of electronics means RESTART, not erase the configuration.
When the ESP restart multiple time in a short time window, it will partially reset the configuration, disabling rules and falling back to Sonoff Basic template. See the FAQ [here](https://tasmota.github.io/docs/FAQ/#frequent-rebootsbootloops)
This is what happens when you press RESET multiple times
It is however possible to change the fallback template in your build if you want.

The whole configuration can be erased by Tasmota through different mecanism listed in the [Recovery](https://tasmota.github.io/docs/Device-Recovery/) section of the doc
Typical reset method when the device as a "Button" is to press the button for 40 seconds.

**Reply by NsinghP** [reply]
I have compiled a bin file online at tasmocompiler after editing (user config override.h) as follows

(1.a) In (user_config_override.h)  add  #define USE_UFILESYS
(1.b) In (user_config_override.h) add   #define GUI_TRASH_FILE

(2) #define MODULE                 USER_MODULE      

(3) #define USER_TEMPLATE "{\"NAME\":\"MYHOME\",\"GPIO\":[258,259,1,163,256,257,1,1,162,161,160,1,1,1],\"FLAG\":0,\"BASE\":18}" 

(5) #define EMULATION              EMUL_HUE         

(6) #define FRIENDLY_NAME          "ROOM1"    

Then I have updated the device to mimimal.bin and then to the new custom.bin file. 

New module MYHOME shows up. Along with Hue enabled. But reset does not ask for wifi credentials.

**Reply by NsinghP** [reply]
let me try the 40 second method.

**Reply by sfromis** [reply]
You can also do a "factory reset" of Tasmota using the command `Reset 1` which will revert to the defaults settings in the binary.

**Reply by NsinghP** [reply]
Reset 1 command from web console is giving desired results and prompting to enter wifi information. However, I still want physical button reset option so that wifi credential prompt is available in case user enters wrong wifi password during the initial setup. I can't seem to figure it out. Is there a command that activates "Button1" option. On my device (ESP8266-12F Bare chip based device) push button is connected to GND and RST pin. Fast Cycle Device recovery procedure is not working. Setoption65 is set to 0. What am I missing?

**Reply by barbudor** [reply]
What is your device ?
Are you using Switch because they are rocker switches ? If they are momentary-push, better to use Button, which would provide you with the Button_1 you need

How did you tried the Fast Cycle Recovery ? You need to actually power off and on the device

**Reply by barbudor** [reply]
You can simulate the Button_1 behavior with a Switch using Rules
As you know how to have filesystem and autoexec.bat you can have rules automatically recreated.
But you still have to connect to the wifi once to have the rules loaded

**Reply by GVollaro** [reply]
I think if you want the device to take the user to Tasmota access point if wifi settings aren't correct, you had to set in my_user_config.h:
#define WIFI_CONFIG_TOOL       WIFI_MANAGER        
// [WifiConfig] Default tool if wifi fails to connect (default option: 4 - WIFI_RETRY)
// (WIFI_RESTART, WIFI_MANAGER, WIFI_RETRY, WIFI_WAIT, WIFI_SERIAL, WIFI_MANAGER_RESET_ONLY)
// The configuration can be changed after first setup using WifiConfig 0, 2, 4, 5, 6 and 7.

**Reply by NsinghP** [reply]
Ok, I am still confused about Button_1 issue. I have ESP8266-12F based device that controls 4 realys ( GPIO-0 --->Relay_i-1, GPIO-4 --->Relay_i-2, GPIO-5 --->Relay_i-3, GPIO-1 --->Relay_i-4 ). There are 4 switches that control the relay (GPIO-14 --->Switch-1, GPIO-13 --->Switch-2, GPIO-12 --->Switch-3, GPIO-3 --->Switch-4 ). On the PCB I have 2-Pin momentary push button switch which is normally open and it connects RST pin to GND. (Red push button on the lower left side of PCB in the attached Image). Can i define behavior of this switch through tasmota?

![ESP8266 device](https://user-images.githubusercontent.com/69470381/114187968-3c298500-9966-11eb-8c5f-ac4fc5c55f35.jpeg)

**Reply by barbudor** [reply]
> Can i define behavior of this switch through tasmota?

NO.
RESET is an ESP pin that restart the ESP. Not under software control

**Reply by barbudor** [reply]
Why are you defining "Switch" instead of "Button" ?

Explain in your how words what is for you the difference between Switch and Button and why you chose Switch

**Reply by NsinghP** [reply]
well in some rooms i use TTP223 based touch modules and in some rooms i use normal on/off houshold AC220V wall switches. and i never thought of defining them as buttons because switch definition has been working and giving the desired results.

**Reply by NsinghP** [reply]
Thanks for your time and effort in answering my questions. I think I understand this Button_1 issue now! What i need to do is either add another push button to an available GPIO (Which may be a problem as I am using 8 GPIO's already and finding  one more that can accept input may not be possible) and define it as a button_1. Or I can define one of the existing switchs as Button_1. 

Please correct me if there is error in my assumption.

**Reply by NsinghP** [reply]
Allright, So on my device GPIO-02 was free. I warmed up my soldering iron and soldered a pushbutton between GPIO-02 and GND. Defined GPIO-02 as Button1 and bingo my problem is solved. When I press this pushbutton for 40 seconds, my device resets and asks for wifi configuration. As an added benefit, since GPIO-02 controls the ESP8266-12f onboard Led light, it glows the entire time this button is pressed letting me know that reset is working. 

I would like to thank everyone here for taking the time to help me, especially barbudor, your knowledge of Tasmota is impressive, thanks for nudging me in the right direction. 

Regards,

Navi

**Reply by NsinghP** [reply]
> 
> How did you tried the Fast Cycle Recovery ? You need to actually power off and on the device

I was doing it wrong. By pressing the HW_Reset button 6 times after turning the power off for a minute!!!
